File: common/src/main/java/dev/obscuria/fragmentum/v2/core/common/registry/_DeferredAttribute.java
package dev.obscuria.fragmentum.v2.core.common.registry;

import dev.obscuria.fragmentum.v2.api.common.registry.DeferredAttribute;
import net.minecraft.core.Holder;
import net.minecraft.world.entity.ai.attributes.Attribute;

public final class _DeferredAttribute extends _Deferred<Attribute> implements DeferredAttribute {

    public _DeferredAttribute(Holder<Attribute> holder) {
        super(holder);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/api/registry/DeferredAttribute.kt
package dev.obscuria.fragmentum.api.registry

import net.minecraft.core.Holder
import net.minecraft.world.entity.ai.attributes.Attribute
import java.util.function.Supplier

class DeferredAttribute(supplier: Supplier<Holder<Attribute>>)
    : Deferred<Attribute, Attribute>(supplier)